From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jerzy Kolosowski <jurek@kolosowscy.pl>
Date: Thu, 10 Apr 2026 12:00:00 +0200
Subject: [PATCH] Revert Use kernel command line to disable memory cgroup

Revert partially the following commit to have
cgroup enabled by default in Fedora/RHEL:

The RPi foundation disables memory cgroups by default via
cgroup_disable=memory in DTS bootargs. This breaks systemd/cgroups v2
on Fedora/CentOS/RHEL.

Rebased from dwrobel/kernel dw-6.12.y for rpi-6.18.y.

Signed-off-by: Jerzy Kolosowski <jurek@kolosowscy.pl>
Based-on-patch-by: Damian Wrobel <dwrobel@ertelnet.rybnik.pl>

---
 arch/arm/boot/dts/broadcom/bcm2708-rpi-bt.dtsi | 2 +-
 arch/arm/boot/dts/broadcom/bcm270x.dtsi        | 2 +-
 arch/arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi | 2 +-
 arch/arm/boot/dts/broadcom/bcm271x-rpi-bt.dtsi | 2 +-
 arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi  | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/arm/boot/dts/broadcom/bcm2708-rpi-bt.dtsi b/arch/arm/boot/dts/broadcom/bcm2708-rpi-bt.dtsi
index 1111111111111..2222222222222 100644
--- a/arch/arm/boot/dts/broadcom/bcm2708-rpi-bt.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm2708-rpi-bt.dtsi
@@ -24,7 +24,7 @@ minibt: bluetooth {

 / {
 	chosen {
-		bootargs = "coherent_pool=1M 8250.nr_uarts=1 snd_bcm2835.enable_headphones=0 cgroup_disable=memory";
+		bootargs = "coherent_pool=1M 8250.nr_uarts=1 snd_bcm2835.enable_headphones=0";
 	};

 	aliases {
diff --git a/arch/arm/boot/dts/broadcom/bcm270x.dtsi b/arch/arm/boot/dts/broadcom/bcm270x.dtsi
index 1111111111111..2222222222222 100644
--- a/arch/arm/boot/dts/broadcom/bcm270x.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm270x.dtsi
@@ -4,7 +4,7 @@
 / {
 	chosen: chosen {
 		// Disable audio by default
-		bootargs = "coherent_pool=1M snd_bcm2835.enable_headphones=0 cgroup_disable=memory";
+		bootargs = "coherent_pool=1M snd_bcm2835.enable_headphones=0";
 		stdout-path = "serial0:115200n8";
 	};

diff --git a/arch/arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi b/arch/arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi
index 1111111111111..2222222222222 100644
--- a/arch/arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi
@@ -3,7 +3,7 @@

 / {
 	chosen {
-		bootargs = "coherent_pool=1M 8250.nr_uarts=1 snd_bcm2835.enable_headphones=0 cgroup_disable=memory numa_policy=interleave nvme.max_host_mem_size_mb=32";
+		bootargs = "coherent_pool=1M 8250.nr_uarts=1 snd_bcm2835.enable_headphones=0 numa_policy=interleave nvme.max_host_mem_size_mb=32";
 	};

 	__overrides__ {
diff --git a/arch/arm/boot/dts/broadcom/bcm271x-rpi-bt.dtsi b/arch/arm/boot/dts/broadcom/bcm271x-rpi-bt.dtsi
index 1111111111111..2222222222222 100644
--- a/arch/arm/boot/dts/broadcom/bcm271x-rpi-bt.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm271x-rpi-bt.dtsi
@@ -24,7 +24,7 @@ minibt: bluetooth {

 / {
 	chosen {
-		bootargs = "coherent_pool=1M 8250.nr_uarts=1 snd_bcm2835.enable_headphones=0 cgroup_disable=memory";
+		bootargs = "coherent_pool=1M 8250.nr_uarts=1 snd_bcm2835.enable_headphones=0";
 	};

 	aliases {
diff --git a/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi b/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi
index 1111111111111..2222222222222 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi
@@ -113,7 +113,7 @@

 / {
 	chosen: chosen {
-		bootargs = "reboot=w coherent_pool=1M 8250.nr_uarts=1 pci=pcie_bus_safe cgroup_disable=memory numa_policy=interleave nvme.max_host_mem_size_mb=32";
+		bootargs = "reboot=w coherent_pool=1M 8250.nr_uarts=1 pci=pcie_bus_safe numa_policy=interleave nvme.max_host_mem_size_mb=32";
 		stdout-path = "serial10:115200n8";
 	};

--
2.47.0

